src/coreclr/jit/valuenum.cpp

```diff
@@ -8550,6 +8550,21 @@ void Compiler::fgValueNumberSsaVarDef(GenTreeLclVarCommon* lcl)
 static bool GetStaticFieldSeqAndAddress(ValueNumStore* vnStore, GenTree* tree, ssize_t* pAddress, FieldSeq** pFseq)
 {
     ssize_t val = 0;
+
+    // Special case for NativeAOT: ADD(ICON_STATIC, CNS_INT) where CNS_INT has field sequence corresponding to field's
+    // offset
+    if (tree->OperIs(GT_ADD) && tree->gtGetOp1()->IsIconHandle(GTF_ICON_STATIC_HDL) && tree->gtGetOp2()->IsCnsIntOrI())
+    {
+        GenTreeIntCon* cns1 = tree->gtGetOp1()->AsIntCon();
+        GenTreeIntCon* cns2 = tree->gtGetOp2()->AsIntCon();
+        if ((cns1->gtFieldSeq == nullptr) && (cns2->gtFieldSeq != nullptr))
```

Comment:
It is sufficient to check that `cns2` has a field sequence. Field sequence + field sequence is invalid IR.